---
layout: default
title: Projects & Open-Source
tags: projects
permalink: /projects/
js:
- js/gh_projects.js
- js/util_buttons.js
---

<!-- <div id="display-projects"></div>

<div class="circle-button">
      <div class="top-circle" data-sr-id="80" style="; visibility: visible;  -webkit-transform: scale(1); opacity: 1;transform: scale(1); opacity: 1;-webkit-transition: -webkit-transform 0.5s cubic-bezier(0.6, 0.2, 0.1, 1) 0s, opacity 0.5s cubic-bezier(0.6, 0.2, 0.1, 1) 0s; transition: transform 0.5s cubic-bezier(0.6, 0.2, 0.1, 1) 0s, opacity 0.5s cubic-bezier(0.6, 0.2, 0.1, 1) 0s; ">
        <img src="{{ site.url }}/images/logos/top.png">
  </div>
</div>

{% for js in page.js %}
  <script type="text/javascript">
  {% include {{ js }} %}
  </script>
{% endfor %}

<script src="https://cdn.jsdelivr.net/emojione/3.0.3/lib/js/emojione.min.js"></script>
<script type="text/javascript">
  $(function() {
    $("#display-projects").getRepos("thundergolfer"); /* Add your github username. */
  });
</script> -->


<h1> Machine Learning </h1>
<h2> BLISS Astronomical Image Cataloging </h2>
<p> Helped develop BLISS variational inference software package to encode astronomical catalogs from deep space images. Specific contributions include generalizing the prior distribution over the parameter space (e.g. source density, galaxy morphology, light source fluxes, etc.) to allow for the generation of multi-bandpass images, adjusting the inherited model to encode these newly-generated multi-bandpass images, imbued the package with the ability to align multi-bandpass images relative to a fixed world-coordinate system, implemented several Gaussian Mixture Model priors for generating catalog-specific colors for our randomly-generated light sources, and more. Tools used: Pytorch Lightning, Poetry, Hydra, Python packages (Pytorch, Numpy, scikit-learn, Seaborn, Pandas, OpenCL, Bokeh, etc.), SQL, PyTest, Github Actions, Git, APIs, etc.
      <p>
      <img width="800" align="center" alt="Screen Shot 2023-07-16 at 6.26.01 PM" src="{{ site.url }}/images/Screen Shot 2023-07-16 at 6.26.01 PM.png"> 
      </p>
</p>
<h2> Predicting Gene Expression Levels From Chromatin Structure Using Graph Neural Networks </h2>
<p> Explored the novel HiRes dataset (Liu et al., 2023) to simultaneously profile the transcriptome alongside the 3-D genome. The HiRes dataset contains simultaneously collected Hi-C and scRNA-seq data collected from adult mouse neurons and murine embryonic cells across various time points during development. This project aimed to profile this dataset, learn a latent representation of the contact matrix data and predict gene expression from known gene newtorks from the genomic contact information. Included data preprocessing of the raw data collected from the Gene Expression Omnibus via JuicerTools at a variety of base pair resolutions, learning a lower-dimensional representation of the Hi-C data at the single-cell and population levels (via t-SNE, spectral embedding) and training a graph convolution network and a graph attention network to predict gene expression of gene networks from the associated Hi-C data. </p>
      <p>
      <img width="800" align="center" alt="Screen Shot 2023-12-07 at 12.38.23 AM.png" src="{{ site.url }}/images/Screen Shot 2023-12-07 at 12.38.23 AM.png"> 
      </p>
Tools include: Pytorch geometric, Hicstraw, GEOparse, Numpy, scikit-learn, etc. See 
<h2> Multi-class classification of Chest X-Ray Classification (2022) </h2>
<p>
Built, trained and tested various deep learning architectures (ResNet, VGG16, DenseNet, etc.) via transfer learning for the purpose of 'diagnosing' patients with either COVID-19, viral pneumonia or bacterial pneumonia based on chest x-rays. Project involved use of TensorFlow and NumPy.
  <p> 
<img width="800" align="center" alt="Screen Shot 2023-06-20 at 6 34 04 PM" src="{{ site.url }}/images/Screen Shot 2023-06-20 at 6.33.49 PM.png">        
  </p>
  <p> 
<img width="800" align="center" alt="Screen Shot 2023-06-20 at 6 34 04 PM" src="{{ site.url }}/images/Screen Shot 2023-06-20 at 6.34.04 PM.png">
  </p>
</p>
<h2> Bivariate normal distribution parameter estimation (2022) </h2>
<p>
Determined the parameters of a hidden dataset drawn from a bivariate normal distribution using expectation-maximization and maximum-likelihood estimation. Performance of either method was compared.
</p>

<h1> Undergraduate Thesis | Systems-level modeling of spatial navigation in rodents: a biological neural network </h1>
<p>
      Synthesized the neurophysiological basis of spatial maps distributed across cortical regions into a neurophysiologically-motivated simulation of an agent navigating a familiar environment in two-dimensional space. Neural network model was simulated via MATLAB and consisted of a system of rate-coded, Hebbian neural populations each encoding for a latent variable relevant to spatial navigation. Tools used: Matlab, Git
      <p>
            <video width="800" align="center" controls>
            <source src="{{ site.url }}/images/SpatEpisModel_Darkness_DriftOverlay.mp4" type="video/mp4">
            </video>
      </p>
      <p>
            <video width="800" align="center" controls>
            <source src="{{ site.url }}/images/SpatEpisModel_ExtendedBD2.mp4" type="video/mp4">
            </video>            
      </p>
</p>

<h1> Assorted BioNN simulations </h1> 
<h2> Retinal shift property </h2> 
<p>
Hebbian neural network simulation of shift property exhibited in retinal cells which contribute to a saturating effect in response. Such a characteristic is useful, for example, when suddenly entering an area of intense light where a more linear pattern would result in diffuse, blurred vision.
  <p>
    <img width="800" align="center" alt="Screen Shot 2023-06-20 at 7 03 18 PM" src="{{ site.url }}/images/Screen Shot 2023-06-25 at 12.44.21 AM.png">
  </p>
</p>
<h2> Recurrent, shunting neural network </h2>
<p>
Recurrent, shunting neural network that reflects the ability of a small collection of neurons to separate a noisy, diverse collection of inputs.
  <p>
    <img width="800" align="center" alt="Screen Shot 2023-06-20 at 7 17 37 PM" src="{{ site.url }}/images/Screen Shot 2023-06-25 at 12.44.03 AM.png">
  </p>
</p>

<h1> Web applications </h1> 
<h2> Instagram clone, EECS 485 </h2> 
<p>
  <p>
    <img width="800" align="center" alt="Screen Shot 2023-06-20 at 7 19 10 PM" src="{{ site.url }}/images/Screen Shot 2023-06-25 at 12.43.46 AM.png">
  </p>
</p>
<h2> Google clone, EECS 485 </h2> 
<p>
  <p>
    <img width="800" align="center" alt="Screen Shot 2023-06-20 at 7 20 21 PM" src="{{ site.url }}/images/Screen Shot 2023-11-26 at 1.01.01 AM.png">
  </p>
</p>
